patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -3683,6 +3691,93 @@
  			}
  			else
  			{
+@@ -16061,7 +_,7 @@
+ 						}
+ 					}
+ 				}
+-				else if ((this.sliding || this.velocity.Y == 0f || flag || this.jumpAgainCloud || this.jumpAgainSandstorm || this.jumpAgainBlizzard || this.jumpAgainFart || this.jumpAgainSail || this.jumpAgainUnicorn || (this.wet && this.accFlipper && (!this.mount.Active || !this.mount.Cart))) && (this.releaseJump || (this.autoJump && (this.velocity.Y == 0f || this.sliding))))
++				else if ((this.sliding || this.velocity.Y == 0f || flag || this.jumpAgainCloud || this.jumpAgainSandstorm || this.jumpAgainBlizzard || this.jumpAgainFart || this.jumpAgainSail || this.jumpAgainUnicorn || DoubleJumpHooks.AnyJumpAgain(this) || (this.wet && this.accFlipper && (!this.mount.Active || !this.mount.Cart))) && (this.releaseJump || (this.autoJump && (this.velocity.Y == 0f || this.sliding))))
+ 				{
+ 					if (this.sliding || this.velocity.Y == 0f)
+ 					{
+@@ -16081,6 +_,8 @@
+ 					bool flag5 = false;
+ 					bool flag6 = false;
+ 					bool flag7 = false;
++					bool cloud = false;
++					bool modDoubleJump = false;
+ 					if (!flag)
+ 					{
+ 						if (this.jumpAgainUnicorn)
+@@ -16108,13 +_,20 @@
+ 							this.jumpAgainSail = false;
+ 							flag6 = true;
+ 						}
+-						else
++						else if (this.jumpAgainCloud)
+ 						{
+ 							this.jumpAgainCloud = false;
++							cloud = true;
++						}
++						// Mod jumps are AFTER vanilla jumps here, maybe modder wants to specify before/after?
++						else if(DoubleJumpHooks.SetNextJump(this))
```

Comment:
Before/after may be necessary. It could be as simple as `myJump.beforeVanilla = true`

Reply:
Should this happen via a virtual property that is evaluated when `AddDoubleJump` is called? Or dynamic (modder can change it any time)?